File: app/src/main/java/us/patient/app/factorydesignpattern/Java/CarFactory.java
package us.patient.app.factorydesignpattern.Java;

public class CarFactory {
    public Car getCar(Car.CarType car){
        switch (car){
            case AADI:
                return new AadiCar();
            case MARUTHI:
                return new MaruthiCar();
            case TATA:
                return new TataCar();
            default:
                return null;
        }
    }
}

File: app/src/main/java/us/patient/app/factorydesignpattern/kotlin/CarFactory.kt
package us.patient.app.factorydesignpattern.kotlin

object CarFactory {
      fun getCar(car:Car.CarType) : Car {
          when(car){
              Car.CarType.AADI -> return AadiCar()
              Car.CarType.MARUTHI -> return MarithiCar()
              Car.CarType.TATA -> return TataCar()
          }
      }
}